src/Shared/TaskLoggingHelper.cs

```diff
@@ -235,9 +235,20 @@ public virtual string GetResourceMessage(string resourceName)
             string resourceString = FormatResourceString(resourceName, null);
             return resourceString;
         }
-#endregion
+        #endregion
+
+        #region Message logging methods
 
-#region Message logging methods
+        /// <summary>
+        /// Returns true if a message of given importance should be logged because it is possible that a logger consuming it exists.
+        /// </summary>
+        /// <param name="importance">The importance to check.</param>
+        /// <returns>True if messages of the given importance should be logged, false if it's guaranteed that such messages would be ignored.</returns>
+        public bool LogsMessagesOfImportance(MessageImportance importance)
+        {
+            return BuildEngine is not IBuildEngine10 buildEngine10
+                || buildEngine10.EngineInterface.LogsMessagesOfImportance(importance);
```

Comment:
Is there a case when it won't be an `IBuildEngine10` internally?

Reply:
Yes, `OutOfProcTaskHostNode` implements only `IBuildEngine3` when building as the 3.5 task host. Also, several mock implementations used in tests don't implement all `IBuildEngine` interfaces.